src/notebooks/visualization_notebook.ipynb

```diff
@@ -1023,7 +1023,20 @@
     "\n",
     "# Save the statistics to a file\n",
     "stats_pd = pd.DataFrame(statistics, index=[0])\n",
-    "stats_pd.to_csv(f\"{plot_save_path}/viz_notebook_statistics.csv\", index=False)"
+    "stats_pd.to_csv(f\"{plot_save_path}/viz_notebook_statistics.csv\", index=False)\n",
+    "\n",
+    "# Ensure the subfolder exists\n",
+    "netcdf_path = f\"{plot_save_path}/netCDF\"\n",
+    "os.makedirs(netcdf_path, exist_ok=True)  # Create the directory if it doesn't exist\n",
+    "\n",
+    "# Save NetCDF files\n",
+    "state_vector_labels.to_netcdf(f\"{netcdf_path}/state_vector.nc\")\n",
+    "prior_ds.to_netcdf(f\"{netcdf_path}/prior.nc\")\n",
+    "posterior_ds.to_netcdf(f\"{netcdf_path}/posterior.nc\")\n",
+    "scale.to_netcdf(f\"{netcdf_path}/scale.nc\")\n",
+    "avkern_ROI.to_netcdf(f\"{netcdf_path}/avkern.nc\")\n",
+    "ds.to_netcdf(f\"{netcdf_path}/ds.nc\")\n",
```

Comment:
Can you rename `ds.nc` to `obs_ds.nc` and `ds_regridded.nc` to`obs_ds_regridded.nc`  for the sake of clarity?